meta-intel-edison-distro/recipes-devtools/ninja/files/ninja.py

```diff
@@ -21,6 +21,7 @@
                 NewArgv.append(i)
 
     if seen:
+        w = os.open(fifoname, os.O_WRONLY|os.O_NONBLOCK)
```

Comment:
But is this `w` used anywhere? You can run `pylint` to see the warning it issues on this line (of course it may be more on the rest).

Reply:
Not by our process, but the child (make/ninja) needs it open non-blocking and childs are opening and closing this. I open this one just to make sure the pipe remains. I don't know if that's a bug in our ninja fork (I think upstream they are still struggling to get the jobserver aligned with make). But without this patch it doesn't work.

Note: the whole jobserver stuff is needed as normally you have make orchestrating submakes. We have bitbake orchestrating submakes and subninjas. With this enabled we never have more gcc's running than threads. Now linux has no problem scheduling more gcc's than threads, but each consumes memory, especially g++ like 1GB. So on my machine 16 HT leads to 4 x 16 x 1GB. I used to have 16GB (now 32GB)...